ui/selectors/permissions.js

```diff
@@ -96,6 +97,27 @@ export function getConnectedSubjectsForSelectedAddress(state) {
   return connectedSubjects;
 }
 
+export function getSubjectsWithPermission(state, permissionName) {
+  const subjects = getPermissionSubjects(state);
+
+  const connectedSubjects = [];
+
+  Object.entries(subjects).forEach(([subjectKey, { permissions }]) => {
```

Comment:
Nit: may as well call this parameter `origin`. That is what it is, and would save reassigning it later on line 112.